### PR TITLE
Only allows admin and commander to update an incident if visibility is restricted

### PR DESCRIPTION
**Pull request by mvilanova**

### CODE REVIEW

File: src/dispatch/incident/views.py [hunk 1]
```diff
@@ -138,6 +138,19 @@ def update_incident(
...
         raise HTTPException(status_code=404, detail="The requested incident does not exist.")
 
+    # we want to provide additional protections around restricted incidents
+    if incident.visibility == Visibility.restricted:
+        # reject if the user isn't an admin or commander
+        if incident.commander.email == current_user.email:
+            return incident
+
+        if current_user.role == UserRoles.admin:
+            return incident
```
Comment (kevgliss) [inline src/dispatch/incident/views.py:148]: I think we want to run the update code here.

**Review by kevgliss — CHANGES_REQUESTED** [review]

File: src/dispatch/incident/views.py (hunk 1 above)
Comment (mvilanova) [reply src/dispatch/incident/views.py:145]: Yup, I just noticed the error. Added a commit that fixes it.

File: src/dispatch/incident/views.py [hunk 2]
```diff
@@ -138,6 +138,16 @@ def update_incident(
     if not incident:
         raise HTTPException(status_code=404, detail="The requested incident does not exist.")
 
+    # we want to provide additional protections around restricted incidents
+    if incident.visibility == Visibility.restricted:
+        # reject if the user isn't an admin or commander
+        if current_user.email != incident.commander.email or current_user.role != UserRoles.admin:
+            return incident
```
Comment (kevgliss) [inline src/dispatch/incident/views.py:145]: We should throw the exception here instead of returning.

Comment (mvilanova) [reply src/dispatch/incident/views.py]: Fixed.

**Review by kevgliss — APPROVED** [review]

**Pull request merged**